src/actions/scanner.js

```diff
@@ -90,18 +90,28 @@ export function scannerDispatch (data) {
         let hash = await keccak(txRequest.data.rlp)
         dispatch(selectAccount(account))
         dispatch(scannedTx(hash, tx))
+        Actions.txDetails()
+        dispatch(resetScanner())
       } else if (txRequest.action === 'signTransactionHash') {
         var details = txRequest.data.details
         details.isSafe = false
         let hash = txRequest.data.hash
         dispatch(selectAccount(account))
         dispatch(scannedTx(hash, details))
+        Actions.txDetails()
+        dispatch(resetScanner())
+      } else if (txRequest.action === 'signData') {
+        let data = txRequest.data.data
+        let hash = await keccak(txRequest.data.data)
```

Comment:
That's how it's done in Parity:
https://github.com/paritytech/parity/blob/2fd44e6ce46a1ac546cd9b52d3dc6130b6a368b1/js/src/views/Signer/components/SignRequest/signRequest.js#L144

I currently see that you're not prepending the text before hashing, should it be done in the UI?